### PR TITLE
[RDBMS] Support Minimum TLS version

### PR DESCRIPTION
**Pull request by ramnov**
**Description**  
This PR adds support to specify Minimum TLS version during server create and update for PostgreSQL and MySQL servers. (This feature is not supported for mariadb servers). 

**Testing Guide**  

az postgres server create --minimal-tls-version TLS1_1
az postgres server update --minimal-tls-version TLS1_2
az mysql server create --minimal-tls-version TLS1_0
az mysql server update --minimal-tls-version TLS1_2

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py [hunk 1]
```diff
@@ -83,6 +83,7 @@ def _complex_params(command_group):
             c.argument('administrator_login', options_list=['--admin-user', '-u'], help='Administrator username for the server. Once set, it cannot be changed.')
             c.argument('administrator_login_password', options_list=['--admin-password', '-p'], help='The password of the administrator. Minimum 8 characters and maximum 128 characters. Password must contain characters from three of the following categories: English uppercase letters, English lowercase letters, numbers, and non-alphanumeric characters.')
             c.argument('ssl_enforcement', arg_type=get_enum_type(['Enabled', 'Disabled']), options_list=['--ssl-enforcement'], help='Enable or disable ssl enforcement for connections to server. Default is Enabled.')
+            c.argument('minimal_tls_version', arg_type=get_enum_type(['TLS1_0', 'TLS1_1', 'TLS1_2', 'TLSEnforcementDisabled']), options_list=['--minimal-tls-version'], help='Set the minimal TLS version for connections to server when SSL is enabled. Default is TLS1_2.', validator=tls_validator)
```
Comment (akamath46297) [inline src/azure-cli/azure/cli/command_modules/rdbms/_params.py:86]: Default is none for now . @QingqingYuan2 could you confirm ?

Comment (QingqingYuan2) [reply src/azure-cli/azure/cli/command_modules/rdbms/_params.py:86]: default is "TLSEnforcementDisabled", if no value passed in from cli.

File: src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_commands.py [hunk 2]
```diff
@@ -108,12 +108,28 @@ def _test_server_mgmt(self, database_engine, resource_group_1, resource_group_2)
...
         public_network_access = 'Disabled'
+        minimal_tls_version = 'TLS1_2'
+        default_minimal_tls_version = 'TLSEnforcementDisabled'
 
         geoGeoRedundantBackup = 'Disabled'
         geoBackupRetention = 20
         infrastructureEncryption = 'Enabled'
         geoloc = 'eastus'
 
+        list_checks = [JMESPathCheck('name', servers[0]),
```
Comment (akamath46297) [inline src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_commands.py:119]: We had disabled a test related to restore due to test issue. I had added it back but it never made through could you merge my changes as well and regenerate the play back files ?

https://github.com/Azure/azure-cli/pull/12777

**Comment by yonzhan** [comment]
add to S172

File: src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_commands.py (hunk 2 above)
Comment (ramnov) [reply src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_commands.py:119]: added back

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py (hunk 1 above)
Comment (ramnov) [reply src/azure-cli/azure/cli/command_modules/rdbms/_params.py]: modified default value

**Pull request merged**